_posts/2021/2021-03-19-the-x-factor.md

```diff
@@ -0,0 +1,59 @@
+---
+title: The X factor or how I learned to stop worrying and love the backend
+layout: post
+date: 2021-03-19
+header_image: public/micronaut-header.jpeg
+header_position: center
+header_overlay: true
+category: on-the-job
+tags: ["agile", "company", "team"]
+authors: ["Paolo"]
+about_authors: ["ppriotto"]
+---
+
+It was autumn 2020. I had been at ePages for almost six years, and always in the same frontend team. You could say I was ready for some change. So when our scrum masters started asking around who could imagine joining a new, cross functional team, I was immediately up for it. Four months later, I want to recapitulate the journey since then.
+
+## Done actually
+
+Haven't we done this before? Yes and no. There have always been cross functional teams responsible for certain areas of the ePages products. As far as I can tell, they're successful and satisfied with their team structure. However, the biggest portion of the current product line has been developed following an "API first" approach, with a big core frontend team doing almost all the UI work. This led to cross-team dependencies on tasks, unavoidable handovers, and an "us vs. them" attitude. We'll take a deeper look at these issues in the following paragraphs.
+
+## The incrementable Hulk
+
+With one frontend-only and multiple backend-only teams, our product managers had a hard time juggling the different priorities in order to minimize the time between API design and UI implementation. This was especially visible in the context of [incremental and iterative software development](../../methods-and-tools/the-secret-of-incremental-and-iterative-software-development/) : if the UX designer needed to adapt the concept during the UI implementation, it could happen that the API needed to be adapted, too. In the worst case, the UI team had to wait for the API change, while the team responsible for the API, which had moved on to the next feature, was forced into a context switch, pausing whatever they were currently working on.
```

Comment:
is it a concious choice to speak of "product managers" instead of "product owners"? (despite that, very strong paragraph, though 🥇

Reply:
Unconscious. Now that you make me aware, I prefer product owners. Also, it's shorter.

I think you've just earned another agile movie pun: Life of PO.